src/Apps/Conversation/Components/Conversation.tsx

```diff
@@ -0,0 +1,142 @@
+import { Box, Flex, Image, Serif } from "@artsy/palette"
+import { Conversation_conversation } from "__generated__/Conversation_conversation.graphql"
+import React from "react"
+import { createFragmentContainer, RelayProp } from "react-relay"
+import { graphql } from "relay-runtime"
+import { MessageFragmentContainer as Message } from "./Message"
+import { Reply } from "./Reply"
+
+interface ItemProps {
+  item: Conversation_conversation["items"][0]["item"]
+}
+
+const Item: React.FC<ItemProps> = props => {
+  const { item } = props
+  if (item.__typename === "Artwork") {
+    return (
+      <Flex width="350px">
+        <Flex height="auto" alignItems="center" mr={2}>
+          <Image src={item.image.url} width="55px" />
+        </Flex>
+        <Flex flexDirection="column" justifyContent="center">
+          <Serif size="2" weight="semibold">
+            {item.artistNames}
+          </Serif>
+          <Serif italic size="2" color="black60" lineHeight={1.3}>
+            {item.title}, {item.date}
+          </Serif>
+        </Flex>
+      </Flex>
+    )
+  } else if (item.__typename === "Show") {
+    // it's a partnerShow
+    return (
+      <Flex width="350px">
+        <Flex height="auto" alignItems="center" mr={2}>
+          <Image src={item.coverImage.url} width="55px" />
+        </Flex>
+        <Flex flexDirection="column" justifyContent="center">
+          <Serif size="2" weight="semibold">
+            {item.fair.name}
+          </Serif>
+        </Flex>
+      </Flex>
+    )
+  } else {
+    return null
+  }
+}
+
+interface ConversationProps {
+  conversation: Conversation_conversation
+  relay: RelayProp
+}
+
+const Conversation: React.FC<ConversationProps> = props => {
+  const { conversation, relay } = props
+  const messageCount = conversation.messages.edges.length
+  return (
+    <Box>
+      {conversation.items.map((i, idx) => (
+        <Item item={i.item} key={idx} />
+      ))}
+      {conversation.messages.edges.map((m, idx) => (
+        <Message
+          message={m.node}
+          initialMessage={conversation.initialMessage}
+          key={m.cursor}
+          isFirst={idx === messageCount - 1}
+        />
+      ))}
+      <Reply conversation={conversation} environment={relay.environment} />
```

Comment:
#minor rather than pass around these kinds of props (current user, Relay environment, mediator, other shared global 'grab bag' stuff), you can do `const { relayEnvironment } = useSystemContext()` calls in the specific component that needs it. That can be cleaner vs. a situation of passing lots of props down explicitly all the time.

Reply:
this lead to some weird issues with `getLinkedRecord`, ended up keeping passing `environment` around for now.